mmv1/third_party/validator/kms_key_ring_iam.go

```diff
@@ -89,16 +93,32 @@ func newKmsKeyRingIamAsset(
 
 func FetchKmsKeyRingIamPolicy(d TerraformResourceData, config *Config) (Asset, error) {
 	// Check if the identity field returns a value
-	if _, ok := d.GetOk("{{key_ring_id}}"); !ok {
+	if _, ok := d.GetOk("key_ring_id"); !ok {
 		return Asset{}, ErrEmptyIdentityField
 	}
 
+	assetNameTemplate := constructKmsKeyRingIAMAssetNameTemplate(d)
+
 	// We use key_ring_id in the asset name template to be consistent with newKmsKeyRingIamAsset.
 	return fetchIamPolicy(
 		NewKmsKeyRingIamUpdater,
 		d,
 		config,
-		"//cloudkms.googleapis.com/{{key_ring_id}}", // asset name
-		"cloudkms.googleapis.com/KeyRing",           // asset type
+		assetNameTemplate,                 // asset name
+		"cloudkms.googleapis.com/KeyRing", // asset type
 	)
 }
+
+func constructKmsKeyRingIAMAssetNameTemplate(d TerraformResourceData) string {
+	assetNameTemplate := "//cloudkms.googleapis.com/{{key_ring_id}}"
+	if val, ok := d.GetOk("key_ring_id"); ok {
+		keyRingID := val.(string)
+		splits := strings.Split(keyRingID, "/")
+		if len(splits) == 3 {
+			assetNameTemplate = fmt.Sprintf("//cloudkms.googleapis.com/projects/%s/locations/%s/keyRings/%s", splits[0], splits[1], splits[2])
+		} else if len(splits) == 2 {
+			assetNameTemplate = fmt.Sprintf("//cloudkms.googleapis.com/projects/{{project}}/locations/%s/keyRings/%s", splits[0], splits[1])
+		}
```

Comment:
Nit: I can't quite tell what would happen if the `key_ring_id` is malformed, and we don't reach either of these conditions. Perhaps a schema validation prevents this?

Reply:
`key_ring_id` is from user input, it can go wrong easily, if neither condition is met, then it uses the original string, and I think the code has tried its best to convert in that case.

Reply:
👍 sounds good